_posts/2019-10-28-MongoDB.md

```diff
@@ -0,0 +1,198 @@
+ 
+We also assume that you have quite a bit of knowledge when using MongoDB as you have probably created a database and have come here looking to secure it. If you have little knowledge in MongoDB, you can continue to read this post and learn the theory behind securing your database, however we strongly recommend taking a course and learning the syntax and how MongoDB works to gain a better understanding. With that being said, let's get into it!
+ 
+*Side note: WoTT is a service that checks and audits your node network including MongoDB deployments.*
+ 
+## Common Risks and How to Mitigate Them
+ 
+### 1. Authentication and Authorization
+ 
+By default, when MongoDB is installed, it does without needing password credentials when accessing the database. It is highly recommended that you add users with login credentials, and an admin user with elevated privelages.
+ 
+Before doing so, you need to enable Access Control. By enabling access control, you enforce user authentication, requiring users to identify themselves before accessing your MongoDB deployment. Assuming you have the mongodb server running, connect to the server using a terminal as the client by executing the command:
+ 
+```
+$ mongo
+```
+ 
+*NB: You probably need to add a **<mark>port--</mark>** and **<mark>host--</mark>** after the mongo command depending on how you set up your deployment*.
+ 
+Now that you are in the database as a client, you must switch to the admin database to add a new user. Simply execute the following command:
+ 
+```
+$ use admin
+```
+ 
+You should see your terminal respond with the line:
+ 
+```
+switched to db admin
+```
+ 
+Now we need to create an admin user, to do so, run the following commands:
+ 
+```
+db.createUser(
+   {
+       user: "admin",
+       pwd: "placeholder",
+       roles [{role: "userAdminAnyDatabase", db: "admin"}, "readWriteANyDatabase"]
+   }
+)
+```
+Make sure to replace placeholder with a strong password, you could alternatively use the command:
+ 
+```
+passwordPrompt()
+```
+ 
+ 
+This creates an admin user that has access to every aspect of any database in your deployment, so be cautious as to who you share that information with. Since MongoDB doesn’t enable authorization/authentication by default, even with the creation of an admin user, users will still have access to all aspects of every database in your deployment without the need to authenticate.
+ 
+To Enable authorization, first disconnect from the mongoshell and stop the mongo server (usually using ctrl^c in the terminal will do the trick). Then access the MongoD configuration file using your favourite editor. We will use $EDITOR for the purpose of this tutorial.
+ 
+```
+$ sudo $EDITOR mongod.conf
+```
+ 
+It might be worth mentioning that your configuration file will be by default in the **<mark>/etc/</mark>** directory. If you specified the location yourself please refer to that directory when running the command.
```

Comment:
Remove the `<mark>` tag here as I don't think it's properly styled. Instead use ticks.